src/Featurizers/ShortGrainDropperFeaturizer.h

```diff
@@ -0,0 +1,230 @@
+#pragma once
+
+#include "../Featurizer.h"
+#include "../Archive.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         ContainerHash
+///  \brief         Hash function for Container type
+///
+template <typename Container>
+struct ContainerHash {
+    std::size_t operator()(Container const& container) const noexcept {
+        std::size_t hash = 0;
+        for (typename Container::value_type const & val : container)
+            hash ^= std::hash<typename Container::value_type>{}(val) + 0x9e3779b9 + (hash << 6) + (hash >> 2);
+        return hash;
+    }
+};
+
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         ShortGrainDropperTransformer
+///  \brief         Return true if a grain needs dropping
+///
+class ShortGrainDropperTransformer : public StandardTransformer<std::vector<std::string>, bool> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType                          = StandardTransformer<std::vector<std::string>, bool>;
+    using GrainsSet                         = std::unordered_set<std::vector<std::string>, ContainerHash<std::vector<std::string>>>;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    explicit ShortGrainDropperTransformer(GrainsSet grainsToDrop);
+    explicit ShortGrainDropperTransformer(Archive &ar);
+
+    ~ShortGrainDropperTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(ShortGrainDropperTransformer);
+
+    bool operator==(ShortGrainDropperTransformer const &other) const;
+
+    void save(Archive &ar) const override;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    GrainsSet const                         _grainsToDrop;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override;
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         ShortGrainDropperEstimator
+///  \brief         Estimator to determine which grain to drop given the
+///                 threshod minPoints calculated by:
+///
+///                 windowSize: the rolling window size.
```

Comment:
>windowSize: the rolling window size. [](start = 20, length = 36)

I would expect these parameters to be documented along with the constructor rather than at the class level.